src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health <= 0);
+  constructor(health, name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimale() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden = false) {
+    super(health, name);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal.hidden === false) {
+      animal.health -= 50;
+    } else {
+      animal.health = 100;
+    }
+
+    if (animal.health <= 0) {
+      animal.deleteAnimale();
```

Comment:
Calling `deleteAnimale` (which should be `deleteAnimal`) inside the `bite` method is not necessary. The `Animal` class should have a separate method that checks and removes dead animals from the `Animal.alive` array, and this method should be called after the `bite` operation if needed.